pkg/storage/s3/s3.go

```diff
@@ -1023,6 +1027,44 @@ func (is *ObjectStorage) CheckBlob(repo string, digest godigest.Digest) (bool, i
 	return true, blobSize, nil
 }
 
+// CheckBlobPresence verifies if a blob is present inside a repository. SHOULD lock from outside.
+func (is *ObjectStorage) CheckBlobPresence(repo string, digest godigest.Digest) (bool, int64, error) {
+	if err := digest.Validate(); err != nil {
+		return false, -1, err
+	}
+
+	blobPath := is.BlobPath(repo, digest)
+
+	binfo, err := is.store.Stat(context.Background(), blobPath)
+	if err != nil {
+		is.log.Error().Err(err).Str("blob", blobPath).Msg("failed to stat blob")
+
+		return false, -1, zerr.ErrBlobNotFound
+	}
+
+	// is a 'deduped' blob?
+	if binfo.Size() == 0 {
```

Comment:
The approach in CheckBlob seems cleaner. `if err == nil && binfo.Size() > 0` return and then check the error, and later you don't need this if block.